great_expectations/data_context/data_context/serializable_data_context.py

```diff
@@ -93,15 +93,10 @@ def _save_project_config(self) -> None:
                     logger.info(
                         f"Saving {len(self.zep_config.datasources)} ZEP Datasources to {config_filepath}"
                     )
-                    yaml.dump(
-                        {
-                            **self.config.commented_map,
-                            **self.zep_config._json_dict(),
-                        },
-                        outfile,
-                    )
-                else:
-                    self.config.to_yaml(outfile)
+                    zep_json_dict: dict[str, JSONValues] = self.zep_config._json_dict()
+                    self.config._commented_map.update(zep_json_dict)
```

Comment:
Earlier on I had used `self.config.commented_map` but it's a property with no setter.

Reply:
Not for this PR but maybe we should have a `config.update_comment_map` method instead of reaching into the private variable?